src/main/webapp/app/overview/exercise-chatbot/exercise-chatbot.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component } from '@angular/core';
+import { faCommentDots } from '@fortawesome/free-solid-svg-icons';
+import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { ChatbotPopupComponent } from './chatbot-popup/chatbot-popup.component';
+import { ExerciseChatWidgetComponent } from 'app/overview/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component';
+import { NavigationStart, Router } from '@angular/router';
+
+@Component({
+    selector: 'jhi-exercise-chatbot',
+    templateUrl: './exercise-chatbot.component.html',
+    styleUrls: ['./exercise-chatbot.component.scss'],
+})
+export class ExerciseChatbotComponent {
+    public chatAccepted = 'false';
+    public buttonDisabled = false;
+    private dialogRef: MatDialogRef<ExerciseChatWidgetComponent> | null = null;
+
+    // Icons
+    faCommentDots = faCommentDots;
+
+    constructor(private dialog: MatDialog, private router: Router) {
+        this.router.events.subscribe((event) => {
```

Comment:
Is there a specific reason why you are using router events subscription here? Won't the check in onNgDestroy solve the problem?

Reply:
Also, you should unsubscribe in onNgDestroy anyways.

Reply:
you are right, this is a shorter solution. I updated it.